src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -832,10 +832,11 @@ public void EndBuild()
                 ShutdownConnectedNodes(false /* normal termination */);
                 _noNodesActiveEvent.WaitOne();
 
-                // Wait for all of the actions in the work queue to drain.  Wait() could throw here if there was an unhandled exception
+                // Wait for all of the actions in the work queue to drain.  _workQueue.Completion.Wait() could throw here if there was an unhandled exception
                 // in the work queue, but the top level exception handler there should catch everything and have forwarded it to the
                 // OnThreadException method in this class already.
                 _workQueue.Complete();
+                Task.WaitAny(_workQueue.Completion);
```

Comment:
Why `WaitAny` when you're passing only one task?

Reply:
Because `WaitAny` is only `Task.WaitXX` I am aware of which does not throw Exception if awaited task ends up in a Failed state. Exceptions during _workQueue DataFlow processing are handled in `ProcessWorkQueue` the only exceptions left this could throw, AFAIK, is `OperationCanceledException` which we shall swallow anyway.

Reply:
Interesting! And very unobvious. Would an explicit try/catch with a comment be more appropriate? Or at least a comment.

(Perf super nit: `WaitAny` takes `params Task[]` so the call allocates an array.)

Reply:
> at least a comment

Yes, please.

Reply:
I have though about it more, and it looks like it shall never throws. It will be better it that unexpected exception will surface by `_workQueue.Completion.Wait();`.